### PR TITLE
feat: filters bar for the backpack v2

### LINKED ISSUE #4588: Add ability to filter and sort by in backpack (M1)
The new filter bar has the ability to filter and sort the grid by and specific criteria

**Acceptance Criteria**
- [ ] Should be able to filter by newest, oldest, rarest, less rare, Name A-Z, Name Z-A
- [ ] Only collectible toggles will hide the wearables that are not NFTs
- [ ] The Collection dropdown will enable the selection of multiple linked collections
- [ ] After selecting one option (if different from previously selected) the grid will automatically reload the results with the current criteria

[Documentation](https://www.notion.so/decentraland/PRD-Backpack-V2-13fd81cca29a45a180af6b73c2ca49b5?pvs=4#6311877cc71548c4ad30624519485207) 
[Figma](https://www.figma.com/file/ZU7m0Tiizgef5X6VoIEzWV/%F0%9F%97%BA%EF%B8%8F-Backpack-2.0-%7C-Explorer-%7C-03%2F23?node-id=7-1126&t=Ai9n9VJ89KdhDdrQ-0)

### LINKED ISSUE #4982: Adapt our Wearables Catalog service for supporting the new endpoint (M1)
Create new method in `LambdasWearablesCatalogService` for supporting the use of the new endpoint:

### New endpoint
```
GET /explorer-service/backpack/:address/wearables
```

### Filtering
```
collectionCategory = ('base-wearable' | 'on-chain' | 'third-party')[]
categories: WearableCategory[]
collectionIds: string[] <- third-party collection Ids
rarity: Rarity
name: string
```

### Sorting
```
orderBy: 'date' | 'rarity' | 'name'
direction: 'ASC' | 'DESC'
```

**NOTE:** Platform team deployed in `peer-ue-2.decentraland.zone` a lamb2 version with a mock response for the new endpoint: https://peer-ue-2.decentraland.zone/explorer-service/backpack/myAddress/wearables

**More info:** https://decentralandteam.slack.com/archives/C04SYGSPUN4/p1681741246092459?thread_ts=1680791534.367149&cid=C04SYGSPUN4

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fix #4588 

New filter bar in the Backpack V2:

image

## How to test the changes?
1. Launch the explorer with `DISABLE_backpack_editor_v1&ENABLE_backpack_editor_v2`.
2. Open the new backpack.
3. Check that the new filter bar appears on the top of the backpack.
4. Use each one of the filter controls and check that the wearables panel is updated correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at 7c70575</samp>

Added a new UI component for backpack filters in the backpack editor HUD. This component allows users to filter their backpack items by category, rarity, collection, sorting, and text search. The component consists of a prefab, a view class, a controller class, and an interface. Modified the existing HUD controller and plugin classes to integrate the new component.

### CODE REVIEW

**Review by lorux0 — CHANGES_REQUESTED** [review]

File: unity-renderer/Assets/DCLServices/Lambdas/LambdasService.cs [hunk 1]
```diff
@@ -124,6 +125,35 @@ internal string GetUrl(string endPoint, params (string paramName, string paramVa
             return url;
         }
 
+        internal string AppendQueryParamsToUrl(string url, params (string paramName, string paramValue)[] urlEncodedParams)
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/DCLServices/Lambdas/LambdasService.cs]: You're right, it should be private. Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackEditorHUDControllerShould.cs [hunk 2]
```diff
@@ -74,6 +81,19 @@ public void SetUp()
                 backpackAnalyticsController,
                 wearableGridController,
                 avatarSlotsHUDController);
+
+            var serviceLocator = ServiceLocatorTestFactory.CreateMocked();
+            Environment.Setup(serviceLocator);
+            IWebRequestAsyncOperation webRequestAsyncOperation = Substitute.For<IWebRequestAsyncOperation>();
+            Environment.i.platform.webRequest.Get(
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackEditorHUDControllerShould.cs]: Right, this is not longer needed. Removed!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersComponentViewTests.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            backpackFiltersComponentView.OnSortByChanged += newSorting => resultSorting = newSorting;
+
+            // Act
+            backpackFiltersComponentView.sortByDropdown.GetOption(optionIndex).isOn = false;
+            backpackFiltersComponentView.sortByDropdown.GetOption(optionIndex).isOn = true;
+
+            // Assert
+            switch (optionIndex)
+            {
+                case 0:
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersComponentViewTests.cs]: Ok, I have separated it into 6 specifics tests 👍

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            wearablesCatalogService.GetThirdPartyCollectionsAsync(Arg.Any<CancellationToken>())
+                                   .Returns(UniTask.FromResult(testCollections));
+
+            // Act
+            backpackFiltersController.LoadCollections();
+
+            // Assert
+            backpackFiltersComponentView.Received(1).LoadCollectionDropdown(
+                testCollections,
+                Arg.Any<WearableCollectionsAPIData.Collection>());
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs]: Correct. Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            // Arrange
+            NftCollectionType testCollectionType = NftCollectionType.None;
+            backpackFiltersController.OnCollectionTypeChanged += (collectionType) => testCollectionType = collectionType;
+
+            // Act
+            backpackFiltersComponentView.OnOnlyCollectiblesChanged += Raise.Event<Action<bool>>(isOn);
+
+            // Assert
+            if (isOn)
+                Assert.IsFalse((testCollectionType & NftCollectionType.Base) == NftCollectionType.Base);
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs]: Thanks for the suggestion! Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            HashSet<string> testThirdPartyColections = null;
+            backpackFiltersController.OnThirdPartyCollectionChanged += thirdPartyCollections => testThirdPartyColections = thirdPartyCollections;
+
+            // Act
+            backpackFiltersComponentView.OnCollectionChanged += Raise.Event<Action<HashSet<string>>>(
+                new HashSet<string> { containsDecentralandCollection ? "decentraland" : "testCollection1", "testCollection2", "testCollection3" });
+
+            // Assert
+            if (containsDecentralandCollection)
+                Assert.IsTrue((testCollectionType & NftCollectionType.OnChain) == NftCollectionType.OnChain);
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs]: True... done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+            if (isOn)
+                Assert.IsFalse((testCollectionType & NftCollectionType.Base) == NftCollectionType.Base);
+            else
+                Assert.IsTrue((testCollectionType & NftCollectionType.Base) == NftCollectionType.Base);
+        }
+
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void SetCollectionsCorrectly(bool containsDecentralandCollection)
```
[the remark this answers is not in the document]
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/Tests/BackpackFiltersControllerShould.cs]: Ok, I have changed these names by the next ones:
```
TriggerOnlyCollectiblesFilterFromViewCorrectly
TriggerCollectionsFilterFromViewCorrectly
TriggerSortByFilterFromViewCorrectly
TriggerSearchFilterFromViewCorrectly
```
I think it's more descriptive now.

Thanks!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/IBackpackFiltersComponentView.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+namespace DCL.Backpack
+{
+    public interface IBackpackFiltersComponentView
+    {
+        event Action<bool> OnOnlyCollectiblesChanged;
+        event Action<HashSet<string>> OnCollectionChanged;
+        event Action<(NftOrderByOperation type, bool directionAscendent)> OnSortByChanged;
+        event Action<string> OnSearchTextChanged;
+
+        void Dispose();
```
Comment (davidejensen) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BackpackEditorHUDV2/IBackpackFiltersComponentView.cs:14]: Why are we creating a Dispose in the interface? Shouldn't we use in the BackpackFiltersComponentView the BaseComponent interface that already has the Disposal method?

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/WearablesHelper/WearablesFetchingHelper.cs [hunk 9]
```diff
@@ -13,7 +14,7 @@ public static class WearablesFetchingHelper
         // TODO: change fetching logic to allow for auto-pagination
         // The https://nft-api.decentraland.org/v1/ endpoint doesn't fetch L1 wearables right now, if those need to be re-converted we should use that old endpoint again and change the WearablesAPIData structure again for that response.
         public const string BASE_FETCH_URL = "https://peer.decentraland.org/lambdas/collections";
-        public const string COLLECTIONS_FETCH_PARAMS = "?sortBy=newest&first=1000"; 
+        public const string COLLECTIONS_FETCH_PARAMS = "?sortBy=newest&first=1000";
```
Comment (davidejensen) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Helpers/WearablesHelper/WearablesFetchingHelper.cs:17]: Could be nice to split the first=1000 parameter to create a constant for the maximum number to help possible future change of this number, but it's just my preference

**Review by davidejensen — APPROVED** [review]
Approved with 2 minor comments

**Review by lorux0 — APPROVED** [review]

**Pull request merged**